test/routes/src/server/routes/t_launchpad.js

```diff
@@ -74,113 +74,163 @@ describe('The Launchpad API endpoint', () => {
       });
     });
 
-    context('when snap does not exist', () => {
-      const caveatId = 'dummy caveat';
+    context('when repo contains valid snapcraft.yaml', () => {
+      const snapName = 'dummy-test-snap';
 
       beforeEach(() => {
         nock(conf.get('GITHUB_API_ENDPOINT'))
           .get('/repos/anaccount/arepo')
           .reply(200, { permissions: { admin: true } });
         nock(conf.get('GITHUB_API_ENDPOINT'))
           .get('/repos/anaccount/arepo/contents/snapcraft.yaml')
-          .reply(200, 'name: test-snap\n');
```

Comment:
Fun fact: the tests were actually passing with `test-snap` (even though macaroon requests were not mocked), because `test-snap` seems to be actually a real registered name in public store, so when tests called real API they got valid macaroon response.